frontend/src/pages/notebookController/screens/server/StopServerModal.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { Button, Modal, ModalVariant } from '@patternfly/react-core';
+import { Notebook } from '../../../../types';
+import { deleteNotebook } from '../../../../services/notebookService';
+import useNamespaces from '../../useNamespaces';
+import useNotification from '../../../../utilities/useNotification';
+import { allSettledPromises } from '../../../../utilities/allSettledPromises';
+
+type StopServerModalProps = {
+  notebooksToStop: Notebook[];
+  onNotebooksStop: (stop: boolean) => void;
+};
+
+const StopServerModal: React.FC<StopServerModalProps> = ({ notebooksToStop, onNotebooksStop }) => {
+  const { notebookNamespace } = useNamespaces();
+  const notification = useNotification();
+  const [isDeleting, setDeleting] = React.useState(false);
+
+  const textToShow = notebooksToStop.length > 1 ? 'all servers' : 'server';
+
+  const isModalShown = notebooksToStop.length !== 0;
+
+  const onClose = () => {
+    onNotebooksStop(false);
+  };
+
+  const stopServers = async (projectName: string, notebooks: Notebook[]): Promise<void> => {
+    return allSettledPromises<Notebook | void>(
+      notebooks.map((notebook) => {
+        const notebookName = notebook.metadata.name || '';
+        if (!notebookName) return Promise.resolve();
+        return deleteNotebook(projectName, notebookName);
+      }),
+    ).then(() => {
+      onNotebooksStop(true);
+    });
+  };
+
+  const handleStopServer = () => {
+    setDeleting(true);
+    stopServers(notebookNamespace, notebooksToStop)
+      .catch((e) => {
+        notification.error(`Error stopping ${textToShow}`, e.message);
+      })
+      .finally(() => {
+        setDeleting(false);
+      });
+  };
```

Comment:
Am I missing something, why not just combine these two functions? Why the extra `stopServers` when `handleStopServer` really is just doing that